sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosCatalog.scala

```diff
@@ -322,9 +422,88 @@ class CosmosCatalog
     }
   }
 
-  @throws(classOf[UnsupportedOperationException])
-  override def renameTable(oldIdent: Identifier, newIdent: Identifier): Unit = {
-    throw new UnsupportedOperationException("renaming table not supported")
+  @tailrec
+  private def deleteViewTable(databaseName: String, viewName: String): Boolean = {
+    logInfo(s"deleteViewTable DB:$databaseName, View: $viewName")
+
+    this.viewRepository match {
+      case Some(viewRepositorySnapshot) =>
+        viewRepositorySnapshot.getLatest() match {
+          case Some(viewDefinitionsSnapshot) =>
+            val lastBatchId = viewDefinitionsSnapshot._1
+            val viewDefinitions = viewDefinitionsSnapshot._2
+
+            viewDefinitions.find(v => v.databaseName.equalsIgnoreCase(databaseName) &&
+              v.viewName.equalsIgnoreCase(viewName)) match {
+              case Some(existingView) =>
+                val updatedViewDefinitionsSnapshot: Array[ViewDefinition] =
+                  (ArrayBuffer(viewDefinitions: _*) - existingView).toArray
+
+                if (viewRepositorySnapshot.add(lastBatchId + 1, updatedViewDefinitionsSnapshot)) {
+                  viewRepositorySnapshot.purge(lastBatchId)
+                  true
+                } else {
+                  deleteViewTable(databaseName, viewName)
+                }
+              case None => false
+            }
+          case None => false
+        }
+      case None =>
+        false
+    }
+  }
+
+  private def getClient: CosmosAsyncClient = {
+    this.client
+  }
+
+  private def tryGetContainerMetadata
+  (
+    databaseName: String,
+    containerName: String
+  ): Option[CosmosContainerProperties] = {
+
+    try {
+      Some(getClient
+        .getDatabase(databaseName)
+        .getContainer(containerName)
+        .read()
+        .block()
+        .getProperties)
+    } catch {
+      case e: CosmosException if isNotFound(e) =>
+        None
+    }
+  }
+
+  private def tryGetViewDefinition(databaseName: String,
+                                   containerName: String) : Option[ViewDefinition] = {
+
+    this.tryGetViewDefinitions(databaseName) match {
+      case Some(viewDefinitions) =>
+        viewDefinitions.find(v => databaseName.equalsIgnoreCase(v.databaseName) &&
+          containerName.equalsIgnoreCase(v.viewName))
+      case None => None
+    }
+  }
+
+  private def tryGetViewDefinitions(databaseName: String) : Option[Array[ViewDefinition]] = {
+
+    this.viewRepository match {
+      case Some(viewRepositorySnapshot) =>
+        viewRepositorySnapshot.getLatest() match {
+          case Some(latestMetadataSnapshot) =>
+            val viewDefinitions = latestMetadataSnapshot._2.filter(v => databaseName.equalsIgnoreCase(v.databaseName))
```

Comment:
cosmos db "ids" and database name is case sensitive.

any reason we are using `equalsIgnoreCase` on the databasename

Reply:
Thanks - fixed